src/aiida/cmdline/commands/cmd_presto.py

```diff
@@ -0,0 +1,133 @@
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""``verdi presto`` command."""
+
+from __future__ import annotations
+
+import pathlib
+import re
+import typing as t
+
+import click
+
+from aiida.cmdline.commands.cmd_verdi import verdi
+from aiida.cmdline.utils import echo
+from aiida.manage.configuration import get_config_option
+
+DEFAULT_PROFILE_NAME_PREFIX: str = 'presto'
+
+
+def default_profile_name():
+    from aiida.manage import get_config
+
+    profile_names = get_config().profile_names
+    indices = []
+
+    for profile_name in profile_names:
+        if match := re.search(r'presto[-]?(\d+)?', profile_name):
+            indices.append(match.group(1) or '0')
```

Comment:
Why do you need to keep all indeces in a list? You only need to track the max value right? That should simplify this function a lot.

Reply:
Also need to account for the fact that there could be no indices yet whatsoever. But sure, we could just track the maximum if it exists. But I don't see how this would simplify the code much. I would still have the for-loop and the conditional with the regex, wouldn't I? I would just update a variable containing the current max index?

Reply:
Yes, that's what I mean. The simplification is to just use a simple variable instead of a list (which you then need to sort etc). Perhaps 'significant simplification' was a bit too strong, but still think it would make the code clearer. Right now, I needed to read the code very closely to then realize that you actually only care about the maximum index, If it exists at all.

Reply:
Not sure if it becomes a lot more readable:
```python
def get_default_presto_profile_name():
    from aiida.manage import get_config

    profile_names = get_config().profile_names
    max_index: int = -1

    for profile_name in profile_names:
        if match := re.search(r'presto[-]?(\d+)?', profile_name):
            index = int(match.group(1) or '0')
            if index > max_index:
                max_index = index

    if max_index == -1:
        return DEFAULT_PROFILE_NAME_PREFIX

    return f'{DEFAULT_PROFILE_NAME_PREFIX}-{max_index + 1}'
```
Do you have a simpler version? If not I will keep the list-based one

Reply:
Hmm, I guess it seems wasteful to me to create unnecessary list, but you're right that it's hard to argue about readability one way or the other. Don't feel strongly either way, sorry for making a fuss.